brigade/core/inventory.py

```diff
@@ -189,6 +209,7 @@ def filter(self, filter_func=None, **kwargs):
             filtered = {n: h for n, h in self.hosts.items()
                         if filter_func(h, **kwargs)}
         else:
+            no_match = uuid.uuid4()
             filtered = {n: h for n, h in self.hosts.items()
-                        if all(h[k] == v for k, v in kwargs.items())}
+                        if all(h.get(k, no_match) == v for k, v in kwargs.items())}
```

Comment:
Shouldn't we assume `None`? So, basically, in the inventory:

```
---
my_host1:
    my_var: null
my_host2: {}
```

Both would pass a filter where I say `inv.filter(my_var=None)`

Reply:
I was going back and forth on that one. I can't really think of a good scenario for this. Just wanted to avoid to match more hosts than the user actually indented. Though it could go either way.

Do you want me to remove it?